examples/src/main/java/io/grpc/examples/routeguide/RouteGuideClient.java

```diff
@@ -140,37 +130,33 @@ public void recordRoute(List<Feature> features, int numPoints) throws Interrupte
     StreamObserver<RouteSummary> responseObserver = new StreamObserver<RouteSummary>() {
       @Override
       public void onNext(RouteSummary summary) {
-        info("Finished trip with {0} points. Passed {1} features. "
-            + "Travelled {2} meters. It took {3} seconds.", summary.getPointCount(),
-            summary.getFeatureCount(), summary.getDistance(), summary.getElapsedTime());
+        onRecordRouteResponse(summary);
```

Comment:
I agree with Carl that these indirections make things harder to read. The verification doesn't have to be perfect in the tests, since applications will need completely different verification.

I am okay with modifying the example code to make it easier to test, but would hope it would be minimal and not distracting. For example, if this `recordRoute` method returned `summary` or even just returned a boolean indicating it was successful, then that would seem enough.